sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py

```diff
@@ -232,9 +232,8 @@ def _iter_contextual_wrapper(self, max_wait_time=None):
             # This is not threadsafe, but gives us a way to handle if someone passes
             # different max_wait_times to different iterators and uses them in concert.
             if max_wait_time:
-                # _timeout to _idle_timeout
-                original_timeout = self._handler._idle_timeout
-                self._handler._idle_timeout = max_wait_time * 1000
+                original_timeout = self._handler._timeout
+                self._handler._timeout = max_wait_time * 1
```

Comment:
This looks like it was a uamqp hack - is there a way we can handle this more naturally?